app/models/job_invocation_composer.rb

```diff
@@ -138,18 +138,11 @@ def targeting_params
     end
 
     def triggering_params
-      raise ::Foreman::Exception, _('Cannot specify both recurrence and scheduling') if api_params[:recurrence].present? && api_params[:scheduling].present?
-
-      if api_params[:recurrence].present?
-        {
-          :mode => :recurring,
-          :cronline => api_params[:recurrence][:cron_line],
-          :end_time => format_datetime(api_params[:recurrence][:end_time]),
-          :input_type => :cronline,
-          :max_iteration => api_params[:recurrence][:max_iteration],
-          :purpose => api_params[:recurrence][:purpose],
-        }
-      elsif api_params[:scheduling].present?
+      if api_params[:recurrence].present? && api_params[:scheduling].present?
+        recurring_mode_params.merge :start_at_raw => format_datetime(api_params[:scheduling][:start_at])
+      elsif api_params[:recurrence].present? && api_params[:scheduling].nil?
+        recurring_mode_params
+      elsif api_params[:recurrence].nil? && api_params[:scheduling].present?
```

Comment:
nil checks won't work with hammer
```
[DEBUG 2021-09-07T10:39:02 API] Params: {
    "job_invocation" => {
                     "inputs" => {                                                                     
            "playbook" => "asd"
        },
                        "ssh" => {},
                 "recurrence" => {},               
                 "scheduling" => {                                                                     
            "start_at" => "2021-09-15T17:00:00+00:00"
        },                                        
        "concurrency_control" => {},              
               "search_query" => "name = test-00.example.com",              
                    "feature" => "ansible_run_playbook",                             
                    "ansible" => {},                                                                   
             "targeting_type" => "static_query"                                                        
    }                                      
}
```

Reply:
I replaced nil checks with `empty?`, it should take care of the problem with future scheduling as well.